crates/cli/src/subcommands/generate/csharp.rs

```diff
@@ -430,69 +431,152 @@ fn autogen_csharp_access_funcs_for_struct(
+                            AlgebraicType::I128 => ("I128.MinValue", "I128.MaxValue"),
+                            AlgebraicType::U128 => ("U128.MinValue", "U128.MaxValue"),
+                            AlgebraicType::I256 => ("I256.MinValue", "I256.MaxValue"),
+                            AlgebraicType::U256 => ("U256.MinValue", "U256.MaxValue"),
+                            AlgebraicType::F32 => ("float.MinValue", "float.MaxValue"),
+                            AlgebraicType::F64 => ("double.MinValue", "double.MaxValue"),
+                            _ => todo!(),
+                        };
+                        BTreeColumn { field_type, csharp_field_name_pascal, csharp_field_type: x, range }
+                    },
                 }
+            })
+            .collect();
 
-                let field = &product_type.elements[col_pos];
-                let field_name = field.name.as_ref().expect("autogen'd tuples should have field names");
-                let field_type = &field.algebraic_type;
-                let csharp_field_name_pascal = field_name.replace("r#", "").to_case(Case::Pascal);
-                // NOTE skipping the btree prefix and the table name from the index name
-                let csharp_index_name = (&idx.index_name[table_name.len() + 7..]).to_case(Case::Pascal);
-                let csharp_field_type = match csharp_field_type(field_type) {
-                    None => continue,
-                    Some(x) => x,
-                };
-                writeln!(output, "public class {csharp_index_name}Index");
+        writeln!(output);
+        writeln!(output, "sealed class {0}Comparer : IComparer<{struct_name_pascal_case}>", index.csharp_index_name);
+        indented_block(output, |output| {
+            writeln!(output, "public int Compare({struct_name_pascal_case}? a, {struct_name_pascal_case}? b)");
+            indented_block(output, |output| {
+                writeln!(output, "if (a == null || b == null) return -1;");
+                for field in &fields {
+                    match field.field_type {
+                        AlgebraicType::Product(_) => {
+                            writeln!(output, "var {0} = a.{0}.CompareTo(b.{0});", field.csharp_field_name_pascal);
+                            writeln!(output, "if ({0} != 0) return {0};", field.csharp_field_name_pascal);
+                        },
+                        AlgebraicType::String => {
+                            writeln!(output, "var {0} = String.Compare(a.{0}, b.{0}, StringComparison.InvariantCulture);", field.csharp_field_name_pascal);
+                            writeln!(output, "if ({0} != 0) return {0};", field.csharp_field_name_pascal);
+                        },
+                        _ => {
+                            writeln!(output, "if (a.{0} < b.{0}) return -1;", field.csharp_field_name_pascal);
+                            writeln!(output, "if (a.{0} > b.{0}) return 1;", field.csharp_field_name_pascal);
```

Comment:
All this shouldn't be necessary, you can just return the result of `String.Compare` as-is.